File: Kotlin/Repositories.kt
package com.kamino.banking.repository

import com.kamino.banking.entity.*
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.data.jpa.repository.Query
import java.time.LocalDateTime

interface TransacaoFinanceiraRepository : JpaRepository<TransacaoFinanceira, Long> {

    @Query("SELECT t FROM TransacaoFinanceira t WHERE t.idContaBanco = :idContaBanco ORDER BY t.data DESC")
    fun findUltimaTransacao(idContaBanco: Int): TransacaoFinanceira?

    @Query("""
        SELECT t FROM TransacaoFinanceira t 
        WHERE t.idContaBanco = :idContaBanco 
        AND t.data BETWEEN :dataInicio AND :dataFim
    """)
    fun findByContaBancoAndPeriodo(
        idContaBanco: Int,
        dataInicio: LocalDateTime,
        dataFim: LocalDateTime
    ): List<TransacaoFinanceira>

    fun findByCodigoNoBanco(codigoNoBanco: String): TransacaoFinanceira?
}

interface ContaBancoRepository : JpaRepository<ContaBanco, Int> {

    @Query("SELECT c FROM ContaBanco c WHERE c.idConfigAppExterno IS NOT NULL")
    fun findContasIntegradas(): List<ContaBanco>
}

interface ContaPagarRepository : JpaRepository<ContaPagar, Int> {
    
    @Query("SELECT c FROM ContaPagar c WHERE c.valor = :valor AND c.pago = false")
    fun findPossivelPagamento(valor: BigDecimal): List<ContaPagar>
}

File: TypeScript/repositories.ts
import { Injectable } from '@nestjs/common';
import { InjectRepository } from '@nestjs/typeorm';
import { Repository, Between, IsNull, Not } from 'typeorm';
import { TransacaoFinanceira, ContaBanco, ContaPagar } from './entities';

@Injectable()
export class TransacaoFinanceiraRepository {
  constructor(
    @InjectRepository(TransacaoFinanceira)
    private readonly repository: Repository<TransacaoFinanceira>,
  ) {}

  async findUltimaTransacao(idContaBanco: number): Promise<TransacaoFinanceira | null> {
    return this.repository.findOne({
      where: { idContaBanco },
      order: { data: 'DESC' },
    });
  }

  async findByContaBancoAndPeriodo(
    idContaBanco: number,
    dataInicio: Date,
    dataFim: Date,
  ): Promise<TransacaoFinanceira[]> {
    return this.repository.find({
      where: {
        idContaBanco,
        data: Between(dataInicio, dataFim),
      },
    });
  }

  async findByCodigoNoBanco(codigoNoBanco: string): Promise<TransacaoFinanceira | null> {
    return this.repository.findOne({
      where: { codigoNoBanco },
    });
  }

  async save(transacao: TransacaoFinanceira): Promise<TransacaoFinanceira> {
    return this.repository.save(transacao);
  }
}

@Injectable()
export class ContaBancoRepository {
  constructor(
    @InjectRepository(ContaBanco)
    private readonly repository: Repository<ContaBanco>,
  ) {}

  async findContasIntegradas(): Promise<ContaBanco[]> {
    return this.repository.find({
      where: { idConfigAppExterno: Not(IsNull()) },
    });
  }
}

@Injectable()
export class ContaPagarRepository {
  constructor(
    @InjectRepository(ContaPagar)
    private readonly repository: Repository<ContaPagar>,
  ) {}

  async findPossivelPagamento(valor: number): Promise<ContaPagar[]> {
    return this.repository.find({
      where: { valor, pago: false },
    });
  }

  async save(conta: ContaPagar): Promise<ContaPagar> {
    return this.repository.save(conta);
  }
}
